src/utils/locale.ts

```diff
@@ -135,3 +136,28 @@ export function localizeNumberString(
     return nonExpoNumString;
   });
 }
+
+/**
+ * This interface is for components that need to determine locale from the lang attribute.
+ */
+export interface LangComponent extends GlobalAttrComponent {
+  /**
+   * BCP 47 language tag for desired language and country format
+   *
+   * **Note**: this prop was added exclusively for backwards-compatibility
+   *
+   * @deprecated set the global `lang` attribute on the element instead.
+   * @mdn [lang](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang)
+   */
+  locale?: string;
+}
+
+/**
+ * This util helps resolve a component's locale.
+ * It will also fall back on the deprecated `locale` if a component implemented this previously.
+ *
+ * @param component
+ */
+export function getLocale(component: LangComponent): string {
+  return component.globalAttributes.lang || component.locale || document.documentElement.lang || "en";
```

Comment:
@jcfranco I wonder if we should do...

```ts
return component.globalAttributes.lang || component.locale || component.el.closest("[lang]")?.lang || "en";
```

Would we want to honor a lang attribute set on a parent element?

Reply:
Does supporting the lang attribute from parent elements would require actively updating the children component if the `lang` attribute on the parent changes? Considering there could be a chance that the parent element might not be a calcite component.

Reply:
I don't think we could support changes to parent elements like that.

Reply:
> I don't think we could support changes to parent elements like that.

Agreed.  I feel like if the lang is changing in a parent or perhaps at the root `<html>` tag level, the app should re-render every component anyway with the new lang, at least that is probably the 99% most common use case.

Reply:
Sounds good. I think we need to document that changes to `lang` attr at the component level will trigger re-render but not at the parent level components.

Reply:
Good points. I have something in the works for https://github.com/Esri/calcite-components/pull/5065/files that would allow us to look up lang and have components react appropriately. It still needs to go through some testing to spot any perf hits. I think it'll be OK since most use cases will not be changing lang often (builders are the only case that come to mind, atm).